pkg/csi/cinder/openstack/openstack_snapshots.go

```diff
@@ -39,10 +39,12 @@ const (
 // CreateSnapshot issues a request to take a Snapshot of the specified Volume with the corresponding ID and
 // returns the resultant gophercloud Snapshot Item upon success
 func (os *OpenStack) CreateSnapshot(name, volID string, tags *map[string]string) (*snapshots.Snapshot, error) {
+	// Force the creation of snapshot even the Volume is in in-use state
 	opts := &snapshots.CreateOpts{
 		VolumeID:    volID,
 		Name:        name,
 		Description: snapshotDescription,
+		Force:       true,
```

Comment:
@jichenjc instead of adding it by default, we have option in CreateSnapshotRequest which allows parameters which are opaque to CO but can be used by driver, we could use that  I suppose, if allow_inuse_volume_snapshot: true, then add force:true, something like that  https://github.com/container-storage-interface/spec/blob/master/spec.md

Reply:
@ramineni - most (if not all) use cases that involve snapshots (at least in Kubernetes) will target volumes that are in-use (PVC is bound and used by a pod).

Given that - I do think the default behavior should be to allow in-use and if for some reason we want to change it - we can leverage the opaque parameters.

Reply:
@vkamra we need to go by default behavior given by cinder , by default cinder doesnt allow creating snapshots for inuse volume, so the same behaviour should reflect here to avoid confusion.

Reply:
@ramineni - will defer to you and others here. I do expect most users in Kubernetes to want to enable this option.

@ramineni @jichenjc - would love to see this move forward. With the current limitation we're unable to leverage snapshots because it is not possible for us to scale down running applications in order to snapshot data.

Reply:
> @jichenjc instead of adding it by default, we have option in CreateSnapshotRequest which allows parameters which are opaque to CO but can be used by driver, we could use that I suppose, if allow_inuse_volume_snapshot: true, then add force:true, something like that https://github.com/container-storage-interface/spec/blob/master/spec.md

Do you mean the user needs to add `allow_inuse_volume_snapshot: true` explicitly in the request?

Reply:
> @ramineni - most (if not all) use cases that involve snapshots (at least in Kubernetes) will target volumes that are in-use (PVC is bound and used by a pod).
> 
> Given that - I do think the default behavior should be to allow in-use and if for some reason we want to change it - we can leverage the opaque parameters.

OpenStack Cinder may have different behaviors for different backends (still waiting for the answer from @jichenjc ), if that's the case, I'd rather we add a config option, leaving decision to the cluster admin.

Reply:
I didn't find detail doc on the flag and from code
https://github.com/openstack/cinder/blob/master/cinder/volume/api.py#L879
looks like here will validate the flag 

then here will not handle this flag to lower layer
https://github.com/openstack/cinder/blob/master/cinder/volume/api.py#L864


so seems only check at API layer and not related to backend...

so maybe we can follow the way @lingxiankong mentioned,
add a configuration here to allow such behavior to be disbaled/enabled ? and previously we disable this so how about default to disable and allow admin to change to enable?

Reply:
>> Do you mean the user needs to add allow_inuse_volume_snapshot: true explicitly in the request? 
@lingxiankong yes

Reply:
> CreateSnapshotRequest

yes, I think it's a reasonable request, I will follow up the way

Reply:
> > > Do you mean the user needs to add allow_inuse_volume_snapshot: true explicitly in the request?
> 
> @lingxiankong yes

I don't think it's a good idea, we shouldn't allow the user to enable/disable a feature that depends on the backend configuration.

Reply:
this is not through backend, instead, through configuration 
which is volumestorageclass, will that be reasonable thing to do?

Reply:
@lingxiankong its independent of backend, and by default cinder doesnt allow creating snapshots for in-use volumes irrespective of backend.